app/views/handlers/list_collector_content.py

```diff
@@ -0,0 +1,82 @@
+from functools import cached_property
+
+from flask import url_for
+
+from app.views.contexts import ListContext
+from app.views.handlers.question import Question
+
+
+class ListCollectorContent(Question):
+    @cached_property
+    def repeating_block_ids(self) -> list[str]:
+        return [
+            block["id"] for block in self.rendered_block.get("repeating_blocks", [])
+        ]
+
+    @cached_property
+    def list_name(self) -> str:
+        return self.rendered_block["for_list"]
+
+    def get_context(self):
+        list_context = ListContext(
+            self._language,
+            self._schema,
+            self._questionnaire_store.answer_store,
+            self._questionnaire_store.list_store,
+            self._questionnaire_store.progress_store,
+            self._questionnaire_store.metadata,
+            self._questionnaire_store.response_metadata,
+        )
+
+        return {
+            **list_context(
+                summary_definition=self.rendered_block["summary"],
+                content_definition=self.rendered_block["content"],
+                section_id=self.current_location.section_id,
+                for_list=self.rendered_block["for_list"],
+                return_to=self._return_to,
+                has_repeating_blocks=bool(self.repeating_block_ids),
+            ),
+        }
+
+    def get_next_location_url(self):
+        if incomplete_block := self.get_first_incomplete_repeating_block_location(
+            repeating_block_ids=self.repeating_block_ids,
+            section_id=self.current_location.section_id,
+            list_name=self.list_name,
+        ):
+            return url_for(
+                "questionnaire.block",
+                list_name=self.list_name,
+                list_item_id=incomplete_block.list_item_id,
+                block_id=incomplete_block.block_id,
+                return_to=self._return_to,
+                return_to_answer_id=self._return_to_answer_id,
+                return_to_block_id=self._return_to_block_id,
+            )
+
+        return self.router.get_next_location_url(
+            self._current_location,
+            self._routing_path,
+            self._return_to,
+            self._return_to_answer_id,
+            self._return_to_block_id,
+        )
+
+    def handle_post(self):
```

Comment:
If we inherit from list collector and didn't override, would it not work?.

Reply:
It would error as is yes, but we easily update `_get_answer_action` to handle question not existing, or you can override the method in this content file to return None.

Reply:
But then list collector inherits from Question class `handle_post()` which expects `form.data` which we don't have/send in `Contents` block so it breaks there and at this point it is probably better to keep overriding `handle_post`?

Reply:
I think there is a higher risk in these diverging. We don't need to make form conditional, you just need to use the power of inheritance to separate out the common methods. The current code is also inconsistent with list collector which again is the risk we are trying to avoid. List collector only does any of these if the collector is complete. Here is does all but `_update_section_completeness` if collector isn't complete. The current code already doesn't revaluate the path.

It could be something like:
```py
        if self._is_list_collector_complete():
            self.questionnaire_store_updater.add_completed_location()
            self._routing_path = self.router.routing_path(
                section_id=self._current_location.section_id,
                list_item_id=self._current_location.list_item_id,
            )
            return super(Question, self).handle_post()
```

Calling Questions super does the trick but can be prone to error if we add an inheritance in the middle of it, so it can also be done by you guy making the content of handle post of block.py a common method you can call here. See what the devs prefer.

This will break a test, but the likely issue lies somewhere else and not here as the logic for handle post bar the form handling should be the same between the normal and content list collectors.

Reply:
This is still outstanding. There is still code that shouldn't exist now.

Reply:
I think I reduced it as much as I could now, without breaking it.